UnitTestCommon/UnitTest.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Performance.SDK.Processing;
+using Microsoft.Performance.Toolkit.Engine;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace UnitTestCommon
+{
+    public static class UnitTest
+    {
+        public static void TestTableBuild(RuntimeExecutionResults runtimeExecutionResults, TableDescriptor tableDescriptor, int expectedCount, bool skipDataAvailableCheck = false)
+        {
+            var isTableDataAvailable = runtimeExecutionResults.IsTableDataAvailable(tableDescriptor) == true;
+
+            if (skipDataAvailableCheck || isTableDataAvailable)
+            {
+                var tableResult = runtimeExecutionResults.BuildTable(tableDescriptor);
+                Assert.IsTrue(tableResult.RowCount == expectedCount);
+                var tableData = tableResult.GetDataForAllRows();

```

Comment:
Will GetDataForAllRows actually get exercised since tableData is a local variable and isn't used? Maybe add another assert here?

Reply:
interesting - yea good thought exercise and thing to check. I am guessing even in Retail builds tableResult.GetDataForAllRows(); will run, but indeed tableData local var indeed might be optimized away. Just looking for that func to return an exception or not

Reply:
Results are inconclusive as the debugger under Release build is being weird. I think it's safer to add an extra assert here so will do so.